packages/vscode-messenger-webview/src/messenger.ts

```diff
@@ -189,17 +245,33 @@ export interface MessengerOptions {
     debugLog?: boolean;
 }
 
-export interface RequestData {
-    resolve: (value: unknown) => void,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    reject: (reason?: any) => void
+/**
+ * Create a CancellationToken that is linked to the given signal.
+ *
+ * @param signal An AbortSignal to create a CancellationToken for.
+ * @returns A CancellationToken that is linked to the given signal.
+ */
+export function createCancellationToken(signal: AbortSignal): CancellationToken {
```

Comment:
Isn't AbortSignal available for Node.js, too? Then we could move this to the common package.

Reply:
I think the node version has a slightly different method signatures. In my case `AbortSignal.abort()` has no `reason`  parameter.  I also not sure how common `AbortSignal` in vs-code extension context is.